### PR TITLE
feat: add story to customIcon (#101)

### PR DESCRIPTION
**Pull request by Ajalle**
image

### CODE REVIEW

File: packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.tsx [hunk 1]
```diff
@@ -2,7 +2,11 @@ import { SvgIcon, SvgIconProps } from "@mui/material";
 import React from "react";
 import { IconName, IconSvgPathShapes } from "./common/iconSvgPathShapes";
 
-export interface CustomIconProps extends SvgIconProps {
+type SvgIconBaseProps = Pick<
+  SvgIconProps,
+  "color" | "fontSize" | "htmlColor" | "titleAccess" | "ref"
```
Comment (Ajalle) [inline packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.tsx:7]: I limited available fields here to allow only ones that affect icon itself. 

Should we allow something like `sx` or `classNames` by default? I personally always prefer stricter codebase, but happy to add more fields or allow all of them.

File: packages/data-explorer-ui/.storybook/main.ts [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
     "@storybook/addon-links",
     "@storybook/addon-essentials",
     "@storybook/addon-interactions",
     "@storybook/addon-mdx-gfm",
   ],
+  typescript: {
+    reactDocgen: "react-docgen-typescript",
+    reactDocgenTypescriptOptions: {
+      // Filter out third-party props from node_modules except @mui packages
+      propFilter: (prop: any) =>
```
Comment (Ajalle) [inline packages/data-explorer-ui/.storybook/main.ts:17]: This change allows us to automatically inject MUI fields to story if a component inherits them (but sadly only default versions, ignoring overrides or custom colors)

File: packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.tsx (hunk 1 above)
Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.tsx:7]: Yeh, we normally do abide by a stricter code base - I guess in this particular instance, we can't know whether some external application using this component would like to utilize other props like [viewBox](https://mui.com/material-ui/api/svg-icon/).
I think we should probably, for this component, allow the flexibility for all SvgIconProps to be passed through? Thoughts?

Comment (Ajalle) [reply packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.tsx:7]: This component supposed to provide set of predefined icons, that are easy to use.
We don't want users to use the component for any kind of custom icons, they should use SvgIcon itself for such cases. 
So I would certainly not include any kind of svg-related props. 
But styling props -- is something to discuss.

Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.tsx:7]: Although, thinking this through, currently this component is really set up for package use only (therefore we currently control the set of "custom" icons - let's leave your change as is - we can add viewBox at later date, if needed.

File: packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Meta, StoryObj } from "@storybook/react";
+import { CustomIcon } from "./customIcon";
+
+const meta = {
+  argTypes: {
+    iconName: {
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/CustomIcon/customIcon.stories.tsx:6]:
```suggestion
    htmlColor: {
      control: "color",
      description: `Applies a color attribute to the SVG element when color is "inherit" or "undefined".`,
    },
    iconName: {
```

**Review by frano-m — APPROVED** [review]
Awesome thanks @Ajalle

**Pull request merged**